api4/user.go

```diff
@@ -532,6 +535,51 @@ func revokeSession(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func autocompleteUsers(c *Context, w http.ResponseWriter, r *http.Request) {
+	channelId := r.URL.Query().Get("in_channel")
+	teamId := r.URL.Query().Get("in_team")
+	term := r.URL.Query().Get("name")
+
+	var autocomplete []*model.User
+	var err *model.AppError
+
+	searchOptions := map[string]bool{}
+
+	hideFullName := !utils.Cfg.PrivacySettings.ShowFullName
+	if hideFullName && !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		searchOptions[store.USER_SEARCH_OPTION_NAMES_ONLY_NO_FULL_NAME] = true
+	} else {
+		searchOptions[store.USER_SEARCH_OPTION_NAMES_ONLY] = true
+	}
+
+	if len(teamId) > 0 {
+		if !app.SessionHasPermissionToTeam(c.Session, teamId, model.PERMISSION_VIEW_TEAM) {
+			c.SetPermissionError(model.PERMISSION_VIEW_TEAM)
+			return
+		}
+
+		autocomplete, err = app.SearchUsersInTeam(teamId, term, searchOptions)
+	} else if len(channelId) > 0 {
+		if !app.SessionHasPermissionToChannel(c.Session, channelId, model.PERMISSION_READ_CHANNEL) {
+			c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+			return
+		}
+
+		autocomplete, err = app.SearchUsersInChannel(channelId, term, searchOptions)
+	} else {
+		// No permission check required
+		autocomplete, err = app.SearchUsersInTeam("", term, searchOptions)
+	}
```

Comment:
Can you use the `app.Autocomplete...` functions instead of the `app.Search...` ones? With the exception of `SearchUsersInTeam` in the else, that is fine